python/rpdk/java/templates/init/guided_aws/StubUpdateHandler.java

```diff
@@ -0,0 +1,129 @@
+package {{ package_name }};
+
+// TODO: replace all usage of SdkClient with your service client type, e.g; YourServiceAsyncClient
+// import software.amazon.awssdk.services.yourservice.YourServiceAsyncClient;
+
+import software.amazon.awssdk.awscore.AwsRequest;
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.core.SdkClient;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class {{ operation }}Handler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<SdkClient> proxyClient,
+        final Logger logger) {
+
+        this.logger = logger;
+
+        final ResourceModel model = request.getDesiredResourceState();
+
+        // TODO: Adjust Progress Chain according to your implementation
+
+        // STEP 1 [initialize a proxy context]
+        return proxy.initiate("{{ call_graph }}::{{ operation }}", proxyClient, model, callbackContext)
+
+            // STEP 2 [TODO: construct a body of a request]
+            .request(Translator::translateToUpdateRequest)
+
+            // STEP 3 [TODO: make an api call]
+            .call(this::updateResource)
+
+            // STEP 4 [TODO: stabilize is describing the resource until it is in a certain status]
+            .stabilize(this::stabilizedOnUpdate)
+            .progress()
+
+            // STEP 5 [TODO: post stabilization update]
+            .then(progress -> postUpdate(progress, proxyClient))
+
+            // STEP 6 [TODO: describe call/chain to return the resource model]
+            .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
```

Comment:
I think this needs to be reworked.  On create, the first API call is special (in 99% of cases) because it actually creates the resource.  Therefore, it makes sense to have a "create" and a "postCreate".  

However, on update there is no real distinguishment between the first API call (the "update") and the rest (the "postUpdate"), and there may be any number of API calls and stabilizations required, and not necessarily in the order shown here.

Reply:
i see this is actually described pretty well below in the comments for postUpdate (side note: I would again link to the handler contract for desired-state stabilization), but the call chain setup does not really make sense